pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java

```diff
@@ -38,11 +40,16 @@
 public class PinotConnection extends AbstractBaseConnection {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(Connection.class);
+  protected static final String[] POSSIBLE_QUERY_OPTIONS = {
+    QueryOptionKey.ENABLE_NULL_HANDLING,
+    QueryOptionKey.USE_MULTISTAGE_ENGINE
+  };
   private org.apache.pinot.client.Connection _session;
   private boolean _closed;
   private String _controllerURL;
   private PinotControllerTransport _controllerTransport;
-  private final boolean _enableNullHandling;
+  private final Map<String, Boolean> _queryOptions = new HashMap<String, Boolean>();
```

Comment:
I think it is nice to generalise the query options but... couldn't values be other things like ints? I would prefer to use Object as the value of the map and change `DriverUtils::createSetQueryOptionString` to be able to deal with, at least, booleans and integers. Strings would be also fine, but they I'm not sure whether they could be used to do SQL injections or not.

Reply:
The current set of options we are interested in are booleans, so adding this will be for possible future needs. Adding more generic support will increase the processing time per statement execution.

Reply:
I don't think the time that we could spend there would be sensible. Anyway, it seems you already changed that to support objects. Thanks